muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java

```diff
@@ -224,14 +231,30 @@ public Response getVops(@PathParam("IDENTIFIER") String studentIdentifierString)
                 if (mandatority == Mandatority.MANDATORY) {
                   numMandatoryCourses++;
                 }
+
+                SchoolDataIdentifier gradingScaleIdentifier = workspaceAssessment.getGradingScaleIdentifier();
+                if (gradingScaleIdentifier == null) {
+                  break;
+                }
+                SchoolDataIdentifier gradeIdentifier = workspaceAssessment.getGradeIdentifier();
+                if (gradeIdentifier == null) {
+                  break;
+                }
+                GradingScale gradingScale = gradingController.findGradingScale(gradingScaleIdentifier);
+                GradingScaleItem gradingScaleItem = gradingController.findGradingScaleItem(gradingScale, gradeIdentifier);
+                if (!"".equals(gradingScaleItem.getName())) {
+                  grade = gradingScaleItem.getName().substring(0, 1);
```

Comment:
Doesn't seem optimal but I guess has to do as the interface has no better alternative for the "grade code".